integration-libs/punchout/guards/punchout-navigation.guard.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import { GuardResult, Router, RouterStateSnapshot } from '@angular/router';
+import { AuthService, CmsActivatedRouteSnapshot } from '@spartacus/core';
+import { combineLatest, map, Observable, tap } from 'rxjs';
+import {
+  PUNCHOUT_ERROR_PAGE_URL,
+  PUNCHOUT_INSPECT_PAGE_URL,
+  PUNCHOUT_REQUISITION_PAGE_URL,
+  PUNCHOUT_SESSION_PAGE_URL,
+  PunchoutNavigationGuardConfig,
+  PunchOutOperation,
+} from '../root/model';
+import { PunchoutStoreService } from '../root/services';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PunchoutNavigationGuard {
+  protected router = inject(Router);
+  protected punchoutStoreService = inject(PunchoutStoreService);
+  protected authService = inject(AuthService);
+
+  protected readonly allowedUrlsForAll: string[] = [
+    PUNCHOUT_SESSION_PAGE_URL,
+    PUNCHOUT_REQUISITION_PAGE_URL,
+    PUNCHOUT_ERROR_PAGE_URL,
+  ];
+  protected readonly allowCxRoutesForEdit: string[] = [
+    'category',
+    'quickOrder',
+    'product',
+    'cart',
+  ];
+  protected readonly allowedUrlsForInspect: string[] = [
+    PUNCHOUT_INSPECT_PAGE_URL,
+  ];
```

Comment:
[NITPICK]
`allowed...` vs `alllow...` vs `allowed...`

For consistency, shall we rename `allowCxRoutesForEdit` to `allowedCxRoutesForEdit`,